platform/ios/src/MGLMapView.mm

```diff
@@ -4629,6 +4622,15 @@ - (void)updateAnnotationViews
         else
         {
             annotationView.center = [self convertCoordinate:annotationContext.annotation.coordinate toPointToView:self];
+            UIView <MGLCalloutView> *calloutView = self.calloutViewForSelectedAnnotation;
+            if (calloutView && calloutView.representedObject == annotationContext.annotation) {
+                CGPoint point = annotationView.center;
+                point.y -= annotationView.bounds.size.height/2.0f;
```

Comment:
Eventually we’ll support shifting the viewport to fit the callout (#1089) and it might be nice to support alternative popup directions (bottom, side, etc), but let’s not worry about those until #4392.